### DISCUSSION TITLE
Is Linux Fedora 28 supported?

### DISCUSSION THREAD

**Discussion by janemao527** [AvaloniaUI/Avalonia#13771] [Q&A]
The document https://docs.avaloniaui.net/docs/faq says Fedora 30+ is supported, we built an application with Avalonia, it can be run in Fedora 28 (Redhat 8.9). 
So my questions are:
1. Is Fedora 28 supported? 
2. If we find any issue on Fedora 28, will it be fixed?

Thanks.

**Comment by maxkatz6** [comment]
1. It can technically work if all dependencies are installed and compatible by themselves. Or might not. We do not forcefully stop supporting older platforms.
2. No. Anything specific to the older version of the OS, that isn't reproducible on the newer version, won't be fixed by the team. Contribution with a fix can be accepted though, if it doesn't complicate maintaining of the code.

**Reply by maxkatz6** [reply]
In general, this document needs to be updated, at least macOS and android/ios versions should be higher than that. But I don't think anything was changed about linux.

**Comment by cmoaciopm** [comment]
@maxkatz6 I have several questions regarding this topic.

- Where does the minimum supported OS come from? Is it just the minimum OS that Avalonia team has verified?
- As you said"_It can technically work if all dependencies are installed and compatible by themselves_", why not listing the dependencies in the document and say as long as the dependencies are satisfied, Avalonia should work? Then you don't need to maintain a list of the supported OSes in the document anymore. People who are interested in Avalonia will not ask the same tedious question about why my Mint/OpenSUSE is not supported?  From the application developer perspective, they will have more confidence to adopt Avalonia as long as their target Linux distribution fulfills the dependencies.
- I'm a little worried about "_Anything specific to the older version of the OS, that isn't reproducible on the newer version, won't be fixed by the team_". Does that mean an issue which can be reproduced on Ubuntu 22.04 but not on Ubuntu 20.04 will not be fixed? If yes, this will make me hesitate to choose Avalonia. Because Ubuntu 20.04 will be end of support in 2025. I can't imagine Avalonia team refuses to fix the issue in a matured, long-term-supported popular Linux distribution.

**Reply by maxkatz6** [reply]
> Does that mean an issue that can be reproduced on Ubuntu 22.04 but not on Ubuntu 20.04 will not be fixed

The opposite. Let's start with the minimal Ubuntu version that we support - Ubuntu 16.04. If any issue is reproducible on any older version than 16.04 - it might never be fixed. If it is reproducible on 16.04 or newer - we still support it, and it might be fixed.

In general, this whole "minimal supported version" is set to limit the unlimited scope of supported devices. But as I mentioned above, we do not intentionally limit supported platforms.

For example, a minimal supported Windows version is 8.1 at this moment. Which aligns with MS support roadmap. But Avalonia still works on Windows 7. The only difference that some features won't be supported there, like window transparency, or newer touch-related APIs. Some of these issues are limited by the platform API itself, and some of them are justified by reducing maintenance burden.

> why not listing the dependencies

Yes, we can do that for this page. Would make much sense.
Also, we need to update minimal supported versions, as it wasn't updated since 0.10.

**Reply by maxkatz6** [reply]
Also, something very important I didn't mentioned above - .NET minimal versions support policy. We can't run Avalonia on something that .NET cannot run. 
Please check out this page: https://github.com/dotnet/core/blob/main/release-notes/8.0/supported-os.md

.NET 8 supports Windows 10+, Fedora 37+, macOS 10.15+. Can it run on older OS? I don't know for all of them, but it surely runs on Windows 7. But .NET team would unlikely to fix any windows 7 related bugs.

**Reply by maxkatz6** [reply]
Which also reminded me. Most of these specific minimal version that Avalonia supports do come from .NET supported versions. Keep in mind that Avalonia works with .NET Core 2.1 at least, which is out of support by itself, but nobody would stop you from using it.

**Reply by maxkatz6** [reply]
After this discussion, I need to update FAQ to include important information about versions supported by .NET. Will do it tomorrow.

**Reply by maxkatz6** [reply]
https://docs.avaloniaui.net/docs/faq#what-platforms-are-supported

**Reply by cmoaciopm** [reply]
Thanks for your update. The updated document makes sense to me.  From the above link, _The main limitation is .NET SDK support and availability of X11 system._, looks like Avalonia should work in most Linux distributions as the prerequisite is quite fundamental.
